Docker documentation update

shiroginne opened the pull request:
# Ticket
No ticked. 

# What are you trying to accomplish?
Added `Migrating Your Local Database Between Docker Versions` section describing how to persist PG data.
Following up https://github.com/opf/openproject/pull/19135

Please let me know if something here is unclear or needs more description. 

# Merge checklist

- [x] Added/updated documentation in Lookbook (patterns, previews, etc)

Kharonus commented on docs/development/development-environment/docker/README.md:
```diff
@@ -537,3 +537,42 @@ This means that the current image is out-dated. You can update it like this:
+1. Start your current environment with the existing database and backend images. Make sure the database service is running normally.
+2. Create a database dump using a PostgreSQL-compatible tool such as `pg_dump` or `pg_dumpall`. This ensures you have a consistent export of your data before making any destructive changes.
+3. Copy the resulting dump file from inside the database container to your local machine, so it remains accessible after removing the Docker volumes.
+> Copying backup to the local machine:
```
🟡 I'd rather not use `>`  in this ordered list. either use normal text, or put it inside the code block with a comment (#)

Kharonus commented on docs/development/development-environment/docker/README.md:
```diff
@@ -537,3 +537,42 @@ This means that the current image is out-dated. You can update it like this:
+
+## Migrating Your Local Database Between Docker Versions
+
+When upgrading your local development stack (e.g., after pulling the latest dev branch or updating Docker images), you may want to preserve your PostgreSQL data. This guide outlines the correct steps to safely migrate your data.
```
🟡 Not sure, how you usually write markdown, but is there a possibility you configure your editor to have line wraps formatted at 120 characters?

Kharonus commented on docs/development/development-environment/docker/README.md:
```diff
@@ -537,3 +537,42 @@ This means that the current image is out-dated. You can update it like this:
+5. Update the codebase by pulling the latest changes from the dev branch. You may also want to update Docker base images at this stage.
+6. Rebuild the backend image to ensure it’s aligned with the current code and dependency versions.
+7. Start only the database service, allowing it to initialize with a clean or migrated volume, depending on your setup.
+8. Copy your previously saved database dump back into the container and restore it into the PostgreSQL instance or load the dump from local machine. This will rehydrate the new database with your old data.
```
🟡 maybe you should add the `docker cp` commands, too?

Kharonus replied:
disregard that. your commands create the dumpfile directly on the host fs.

Kharonus commented on docs/development/development-environment/docker/README.md:
````diff
@@ -537,3 +537,42 @@ This means that the current image is out-dated. You can update it like this:
+```shell
+docker compose exec -T <db-container-name> pg_dumpall -U <db-user> <db-name> > openproject_full_backup.sql
+```
+4. Shut down the Docker stack, including all services and volumes. If you want a clean reset, make sure to remove the database volume. If you’re only updating images and the volume is compatible, this step can be adjusted accordingly.
````
not sure, I understand. Why would you NOT drop the database volume when using that guide? If you do not drop volume, the import will also not be working, right?

shiroginne replied:
true, I've changed it to:
> 4. Shut down the Docker stack. If you want a clean reset, make sure to remove the database volume.

Kharonus commented on docs/development/development-environment/docker/README.md:
````diff
@@ -537,3 +537,42 @@ This means that the current image is out-dated. You can update it like this:
+```
+or
+```shell
+docker compose exec -T <db-container-name> pg_dumpall -U <db-user> <db-name> > openproject_full_backup.sql
````
🔴 Correct me if I'm wrong, but dump_all does not take a db-name as argument. The correct line would be:

```
docker compose exec -T <db-container-name> pg_dumpall -U <db-user> > openproject_full_backup.sql
```

shiroginne replied:
Yes, true. Copy-paste mistake 😓

shiroginne committed:
Docker documentation update

Added `Migrating Your Local Database Between Docker Versions` section describing how to persist PG data.

shiroginne commented:
@Kharonus I've just updated the documentation; please take a look

Kharonus reviewed (approved):
LGTM
